### PR TITLE
Adding checks on state_bounds argument to find unexpected keys

### PR DESCRIPTION
**Pull request by andrewlee94**
## Summary/Motivation:
A couple of people have noted that the generic properties framework does not inform the user if they mis-type an entry in the `state_bounds` configuration dict. This PR adds checks for unexpected keys and raises a `ConfigurationError` if one is found (or  logs a warning in the case of mole fractions if they are a state variable).

## Changes proposed in this PR:
-
-

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

**Review by eslickj — APPROVED** [review]
Looks good to me.

File: idaes/generic_models/properties/core/state_definitions/FPhx.py [hunk 1]
```diff
@@ -39,6 +44,23 @@ def define_state(b):
     # TODO: should have some checking to make sure developers implement this properly
     b.always_flash = True
 
+    # Check that only necessary state_bounds are defined
+    expected_keys = ["flow_mol", "enth_mol", "temperature", "pressure"]
```
Comment (jghouse88) [inline idaes/generic_models/properties/core/state_definitions/FPhx.py:48]: Question: Why is `temperature` an expected key when the state definition is `FPhx`?

File: idaes/generic_models/properties/core/state_definitions/FcPh.py [hunk 2]
```diff
@@ -40,6 +45,18 @@ def define_state(b):
     # TODO: should have some checking to make sure developers implement this properly
     b.always_flash = True
 
+    # Check that only necessary state_bounds are defined
+    expected_keys = ["flow_mol_comp", "enth_mol", "temperature", "pressure"]
```
Comment (jghouse88) [inline idaes/generic_models/properties/core/state_definitions/FcPh.py:49]: Same as previous: Why `temperature`?

**Review by jghouse88 — APPROVED** [review]
Minor comment. LGTM.

File: idaes/generic_models/properties/core/state_definitions/FPhx.py (hunk 1 above)
Comment (andrewlee94) [reply idaes/generic_models/properties/core/state_definitions/FPhx.py:48]: Temperature is there to allow users to provide bounds for T as they are generally better known than bounds on enthalpy. These are also very helpful (if not necessary) for initializing the flash calculations.

**Pull request merged**